src/psyclone/psyir/symbols/symbol_table.py

```diff
@@ -56,6 +56,19 @@
 from psyclone.psyir.symbols.typed_symbol import TypedSymbol
 
 
+class _Default:
```

Comment:
I don't think this class is necessary, the implementation below also passes the tests:
```
-class _Default:
-    '''
-    A mutable class used to provide a unique default value to methods within
-    the SymbolTable class. This enables us to determine when the user has
-    supplied a value of 'None' for arguments that have a default value.
-
-    :param Any value: a value associated with this instance.
-
-    '''
-    def __init__(self, value):
-        self.value = value
-
+DEFAULT_SENTINEL = object()
```

```
     def lookup(self, name, visibility=None, scope_limit=None,
-               default=_Default(1)):
+               default=DEFAULT_SENTINEL):
```
```
-            if default is SymbolTable.lookup.__defaults__[-1]:
+            if default is DEFAULT_SENTINEL:
```

Reply:
Embarrassing! The dangers of stack overflow :-)